src/Form/Traits/LinkFieldGetOwnerTrait.php

```diff
@@ -27,8 +27,12 @@ private function getOwnerFields(): array
         $relation = $this->getName();
         // Elemental content block
         if (class_exists(BaseElement::class) && is_a($owner, BaseElement::class)) {
+            // Remove namespaces from inline editable blocks
+            // This will return an empty array for non-inline editable blocks (e.g. blocks in a gridfield)
             $arr = ElementalAreaController::removeNamespacesFromFields([$relation => ''], $owner->ID);
-            $relation = array_keys($arr)[0];
+            if (array_key_exists(0, $arr)) {
+                $relation = array_keys($arr)[0];
+            }
```

Comment:
```suggestion
            if (!empty($arr)) {
                $relation = array_keys($arr)[array_key_first($array)];
            }
```
This feels more robust - without this it seems like there's something magic going on where there might be other values in the array, but we only want values if it's 0-indexed, or something? 

But if we just want the first element (if there are any elements), this code should get us there without ambiguity.

Reply:
Updated, though using slightly different `$relation = array_key_first($arr);` which is equivalent to how it was before

Reply:
Yup, nice. I wasn't looking too closely, my code was garbage lol. Glad you got the idea.